### PR TITLE
[OTel] PubSub & Cloud Storage instrumentation

### PR DESCRIPTION
**Pull request by CagriYonca**
- Pubsub coverage: 90%
- Cloud Storage coverage: 91%

### CODE REVIEW

File: src/instana/instrumentation/google/cloud/storage.py [hunk 1]
```diff
@@ -46,108 +50,139 @@ def _collect_tags(api_request):
...
-            return wrapped(*args, **kwargs)
-
-        with tracer.start_active_span('gcs', child_of=parent_span) as scope:
-            for (k, v) in tags.items():
-                scope.span.set_tag(k, v)
+        parent_context = parent_span.get_span_context() if parent_span else None
 
+        with tracer.start_as_current_span("gcs", span_context=parent_context) as span:
             try:
+                tags = _collect_attributes(kwargs)
```
Comment (GSVarsha) [inline src/instana/instrumentation/google/cloud/storage.py:74]:
```suggestion
                attributes = _collect_attributes(kwargs)
```

**Review by GSVarsha — COMMENTED** [review]
A few minor requests. If there are other higher priority tasks, please feel free to make these changes later as well.

**Review by pvital — CHANGES_REQUESTED** [review]
In addition to @GSVarsha's requests, I have only one that should be applied to multiple files.

File: src/instana/instrumentation/google/cloud/storage.py (hunk 1 above)
Comment (CagriYonca) [reply src/instana/instrumentation/google/cloud/storage.py:74]: Done

File: tests/clients/test_google-cloud-pubsub.py [hunk 2]
```diff
@@ -36,89 +38,91 @@ def setUp(self):
...
-        self.assertEqual(2, len(spans))
-        self.assertIsNone(tracer.active_span)
-        self.assertEqual('gcps', gcps_span.n)
-        self.assertEqual(2, gcps_span.k)  # EXIT
+        assert len(spans) == 2
+
+        current_span = get_current_span()
+        assert not current_span.is_recording()
+        assert gcps_span.n == "gcps"
+        assert gcps_span.k.value == 2
```
[the remark this answers is not in the document]
Comment (CagriYonca) [reply tests/clients/test_google-cloud-pubsub.py:63]: Done

File: src/instana/instrumentation/google/cloud/pubsub.py [hunk 3]
```diff
@@ -2,38 +2,47 @@
...
 import wrapt
-from opentracing import Format
 
-from ....log import logger
-from ....singletons import tracer
-from ....util.traceutils import get_tracer_tuple, tracing_is_off
+from instana.log import logger
+from instana.propagators.format import Format
+from instana.singletons import tracer
+from instana.span.span import InstanaSpan
```
[the remark this answers is not in the document]
Comment (CagriYonca) [reply src/instana/instrumentation/google/cloud/pubsub.py:11]: Done

File: src/instana/instrumentation/google/cloud/pubsub.py [hunk 4]
```diff
@@ -2,38 +2,47 @@
...
+from instana.util.traceutils import get_tracer_tuple, tracing_is_off
 
 try:
     from google.cloud import pubsub_v1
 
-
-    def _set_publisher_tags(span, topic_path):
-        span.set_tag('gcps.op', 'publish')
+    def _set_publisher_attributes(
+        span: InstanaSpan,
```
[the remark this answers is not in the document]
Comment (CagriYonca) [reply src/instana/instrumentation/google/cloud/pubsub.py:18]: Done

**Review by GSVarsha — APPROVED** [review]
LGTM!

**Pull request merged**